plugins/module_utils/github.py

```diff
@@ -1126,6 +1131,15 @@ def _manage_repository(self, state, current=None, check_mode=False, **kwargs):
         owner = kwargs.pop('owner')
         repo_name = kwargs.pop('name')
         current_repo = current if current else self.get_repo(owner, repo_name, ignore_missing=True)
+
+        # When specifying a default branch on a repository which is still empty, the execution
+        # fails with the following error message:
+        # 'Cannot update default branch for an empty repository. Please init the repository and push first'
+        # Therefore we remove this option in that case.
+        if current_repo['size'] == 0:
+            del kwargs['default_branch']
```

Comment:
In that case it might be a good idea to add a notice/warning to get the user aware of that.
How can i do that in Ansible?

Reply:
Too bad, I hoped that there is a way out of Ansible to return a hint or better a warning to the user. 

Originally, the removal of the ‘default_branch’ option was only implemented in the ‘create_repo’ case, but then I decided otherwise because this is a problem if ‘empty’ repositories already exist (‘update_repo’) and another repo makes it necessary to run the Github Manager. In this case, the Github Manager user would then receive an error message that the parameter is not accepted and would then be blocked from further execution.

From a functional point of view, I consider the removal of the parameter to be ‘correct’ as long as the repository does not exist and does not contain a commit. From the point at which commits exist and the default branch does not exist, it is functionally correct to throw an error here.

It's a pity that Github has made the design of the API a bit strange here.